src/EnergyPlus/Construction.cc

```diff
@@ -215,47 +215,48 @@ void ConstructionProps::calculateTransferFunction(EnergyPlusData &state, bool &E
         // Obtain thermal properties from the Material derived type
 
         auto *thisMaterial = state.dataMaterial->Material(CurrentLayer);
+        auto *thisMaterialChild = dynamic_cast<Material::MaterialChild *>(thisMaterial);
```

Comment:
You can keep calling this `thisMaterial`. That will minimize the changes and be shorter also.